pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -308,3 +338,46 @@ func (r *CloudConfigReconciler) setDegradedCondition(ctx context.Context) error
 	klog.Info("Cloud Config Controller is degraded")
 	return r.syncStatus(ctx, co, conds)
 }
+
+func (r *CloudConfigReconciler) setUpgradeableCondition(ctx context.Context, condition configv1.ConditionStatus, reason string, message string) error {
+	co, err := r.getOrCreateClusterOperator(ctx)
+	if err != nil {
+		return err
+	}
+
+	conds := []configv1.ClusterOperatorStatusCondition{
+		newClusterOperatorStatusCondition(configv1.OperatorUpgradeable, configv1.ConditionFalse, reason, message),
+	}
+
+	co.Status.Versions = []configv1.OperandVersion{{Name: operatorVersionKey, Version: r.ReleaseVersion}}
+	if condition == configv1.ConditionFalse {
+		klog.Info("Cloud Config Controller is not upgradeable")
+	} else {
+		klog.Info("Cloud Config Controller is upgradeable")
+	}
+	return r.syncStatus(ctx, co, conds)
+}
+
+// isProviderNutanixAndCloudConfigNeeded will determine whether the proper cloud configuration ConfigMap is present
+// on the Nutanix platform. This function is being added to mitigate an upgrade issue where the user must create
+// the ConfigMap before the upgrade can progress.
+// See https://issues.redhat.com/browse/OCPBUGS-7898 for more information.
+func (r *CloudConfigReconciler) isProviderNutanixAndCloudConfigNeeded(ctx context.Context, platformStatus *configv1.PlatformStatus) (bool, error) {
+	if platformStatus == nil || platformStatus.Type != configv1.NutanixPlatformType {
+		return false, nil
+	}
+
+	cm := &corev1.ConfigMap{}
+	// the name of this ConfigMap is sourced from the Nutanix asset for the volume mount
+	// See: https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/release-4.13/pkg/cloud/nutanix/assets/cloud-controller-manager-deployment.yaml#L107
+	if err := r.Get(ctx, client.ObjectKey{Name: syncedCloudConfigMapName, Namespace: r.ManagedNamespace}, cm); errors.IsNotFound(err) {
```

Comment:
Hm, will this cause issues on first reconciliation when cloud config is not yet synced? Code which calls it runs before the config sync logic. Or customer should be place config into ccm namespace directly?

Reply:
i don't think so because this is looking in the same namespace as the operator. if it doesn't find the configmap it will mark upgradeable=false, but on the next reconcile if the cm exists it will go to true.